### ISSUE TITLE
Variable name conflicts when using `inlineDynamicImports`

### ISSUE THREAD

**Issue by philipwalton** [rollup/rollup#3245]
- Rollup Version: 1.27.2
- Operating System (or Browser): Chrome 78
- Node Version: 10

I'm using Rollup and the `inlineDynamicImports` option to bundle my code, and I've run into the following error:

When Rollup processes the following code:
 
```js
const initLog = async () => {
  const log = await import('./log');
  log.init();
};
```

It turns it into this:

```js
const initLog = async () => {
  const log = await Promise.resolve().then(function () { return log; });
  log.init();
};
```

But notice how in the second line, the function is returning the variable `log` from the outer scope. However, since the function also defines a variable named `log` in the inner scope, instead of returning the imported module, `undefined` gets returned and there's a runtime error.

Note: I'm happy to put together a repro if this isn't enough to go off of.

**Comment by lukastaegert** [comment]
A repro would be nice as I am really confused as to why there is no definition of log in the inner module. Or is it a global variable? In that case, yes, the outer log should have been deconflicted.

**Comment by lukastaegert** [comment]
Ah sorry, `log` is of course the namespace. Yes, a repro would help, when I tried it locally, it was always properly deconflicted.

**Comment by philipwalton** [comment]
Repro: https://github.com/philipwalton/rollup-3245-repro

**Comment by lukastaegert** [comment]
Ah, that is helpful, thanks a lot! I'm starting to get an idea of what is going on here.

**Comment by lukastaegert** [comment]
Using circular imports, it is even reproducible without dynamic imports: https://rollupjs.org/repl/?version=1.27.4&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCcuJTJGbGliMS5qcyclNUNuaW1wb3J0JTIwJy4lMkZsaWIyLmpzJyUzQiU1Q25jb25zb2xlLmxvZygnbWFpbicpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybGliMS5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAnLiUyRmxpYjIuanMnJTNCJTVDbiUyMCUyMCU1Q25jb25zdCUyMG1haW4lMjAlM0QlMjBhc3luYyUyMCgpJTIwJTNEJTNFJTIwJTdCJTVDbiUyMCUyMGNvbnN0JTIwbGliMiUyMCUzRCUyMGF3YWl0JTIwaW1wb3J0KCcuJTJGbGliMi5qcycpJTNCJTVDbiUyMCUyMGNvbnNvbGUubG9nKGxpYjIpJTNCJTVDbiU3RCUzQiU1Q24lNUNubWFpbigpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmxpYjIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJy4lMkZsaWIxLmpzJyUzQiU1Q25leHBvcnQlMjBjb25zdCUyMGZvbyUyMCUzRCUyMDQyJTNCJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzbSUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

The problem is that the namespace object is only deconflicted at the top level but not inside functions that contain dynamic imports that (potentially) reference it.

**Comment by lukastaegert** [comment]
Fix at #3256
